specification.md

```diff
@@ -1152,6 +1159,124 @@ The following attributes SHOULD be included:
 - any user-defined attributes that the end user would like to associate with
   this Agent.
 
+##### AgentDescription.component_details
+
+Details about the available components in the agent.
+
+This field gives a description of what components are available, as well
+as extra metadata about the components.
+
+The structure of ComponentDetails DOES NOT need to be a 1-to-1 match with
+the ComponentHealth structure. ComponentHealth generally refers to instances of
+components, while ComponentDetails refers to the available types of components.
+
+This field is not required to be specified.
+
+#### ComponentDetails Message
+
+Status: [Beta]
+
+The ComponentDetails message has the following structure:
+
+```protobuf
+message ComponentDetails {
+    map<string, ComponentDetails> sub_component_map = 1;
+    repeated KeyValue metadata = 2;
+}
+```
+
+##### ComponentDetails.metadata
+
+Extra key/value pairs that may be used to describe the component.
+
+The key/value pairs are according to semantic conventions, see
+the [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/) for more information.
+
+##### ComponentDetails.sub_component_map
+
+A map of component ID to sub components details. It can nest as deeply as needed to
+describe the underlying system.
+
+##### Examples
+
+###### OpenTelemetry Collector
+
+Here is an example of how ComponentDetails could hold information regarding the included
+components for a custom build collector:
+
+```json
+{
+  "sub_component_data": {
+    "receivers": {
+      "sub_component_data": {
+        "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver@v0.107.0": {
+          "metadata": {
+            "type": "hostmetrics",
+          }
+        }
+      }
+    },
+    "processors": {
+      "sub_component_data": {
+        "go.opentelemetry.io/collector/processor/batchprocessor@v0.107.0": {
+          "metadata": {
+            "type": "batch",
+          }
+        },
+        "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor@v0.107.0": {
+          "metadata": {
+            "type": "transform",
+          }
+        },
+      }
+    },
+    "exporters": {
+      "sub_component_data": {
+        "go.opentelemetry.io/collector/exporter/nopexporter@v0.107.0": {
+          "metadata": {
+            "type": "nop",
+          }
+        }
+      }
+    }
+  }
+  // ... Component list continues for extensions and collectors ...
+}
+```
+
+###### Fluent Bit
+
+Here's an example of how Fluent Bit could report what components it has available.
+
+```json
+{
+  "sub_component_data": {
+    "input": {
+      "sub_component_data": {
+        "tail": {}
+      }
+    },
```

Comment:
For most other agents, I expect this map to be very sparse and probably not useful. Each version in fluent-bit, for example, comes with fixed components at fixed versions (so here, the "version" of the component is just the version of fluent-bit, so it's not repeated here).

This area would be more useful for agents that vary in components they include, so agents like the opentelemetry agent (with the builder), or for agents that have a plugin system for dynamic components.